### PR TITLE
[13.0][ADD] account_asset_management_mass_removal

### PR DESCRIPTION
**Pull request by Saran440**
This Module manages remove the assets by batch process.
it will help you to remove many asset in 1 time.

### CODE REVIEW

**Review by HaraldPanten — CHANGES_REQUESTED** [review]
Use case:

1- Create a new asset with several depreciation lines. Validate all of them.

2- When creating a removal for the next period of the last posted depreciation, an Indexerror occurs:

`Odoo Server Error
Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 624, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 14, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 669, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 350, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 339, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 915, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 1326, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/.repo_requirements/odoo/addons/web/controllers/main.py", line 1314, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 387, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 374, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/build/OCA/account-financial-tools/account_asset_management_removal/models/account_asset_removal.py", line 117, in remove
    {"active_id": asset_line.asset_id.id, "early_removal": True}
  File "/home/odoo/build/OCA/account-financial-tools/account_asset_management/wizard/account_asset_remove.py", line 153, in remove
    residual_value = self._prepare_early_removal(asset)
  File "/home/odoo/build/OCA/account-financial-tools/account_asset_management/wizard/account_asset_remove.py", line 240, in _prepare_early_removal
    first_to_depreciate_dl = dlines[0]
  File "/.repo_requirements/odoo/odoo/models.py", line 5624, in __getitem__
    return self.browse((self._ids[key],))
IndexError: tuple index out of range`

Could you check?

**Comment by Saran440** [comment]
@HaraldPanten Thank you for your reviewed. I fixed it up.  Could you check again?

**Review by HaraldPanten — APPROVED** [review]
It LGTM.

Just one question. Which is the use case that you need to cover with this module? Wouldn't it b enough to make a reversal entry or cancel entries?

**Comment by Saran440** [comment]
okay, I will add test script.

**Comment by HaraldPanten** [comment]
> okay, I will add test script.

Hi @Saran440 . Maybe I didn't explain me well enough.

I mean... You created this new module because you want to cover several use cases.

I'd like to know, what are this daily use cases that you need to cover. The reason why you created this module (account_asset_management_removal). Aren't the Odoo standard reverse entries enough? Are you used to create lots of reversal entries for assets depreciations?

Could you explain me some use cases that you need to cover?

THX

**Comment by Saran440** [comment]
@HaraldPanten Sorry, I misunderstand.

In my case, Company bought many asset on the same day and same as length of period.
when we run depreciation all of them. we want to remove all asset that we will remove 1 by 1.

This module help remove many asset into 1 times for reduct time.
if you want to remove 40 assets, you could created 1 asset removal and add 40 lines (asset) or import.

**Comment by pedrobaeza** [comment]
The module name should be `account_asset_management_mass_removal` or similar, as if not, it can mislead users thinking that the removal is not included in base module and they need this one for that feature.

**Comment by HaraldPanten** [comment]
> @HaraldPanten Sorry, I misunderstand.
> 
> In my case, Company bought many asset on the same day and same as length of period.
> when we run depreciation all of them. we want to remove all asset that we will remove 1 by 1.
> 
> This module help remove many asset into 1 times for reduct time.
> if you want to remove 40 assets, you could created 1 asset removal and add 40 lines (asset) or import.

Sounds useful. Thanks for the explanation.

> The module name should be `account_asset_management_mass_removal` or similar, as if not, it can mislead users thinking that the removal is not included in base module and they need this one for that feature.

I agree. `account_asset_management_mass_removal` sounds more specific 👍

**Comment by luc-demeyer** [comment]
+1 for rename to account_asset_management_mass_removal.

**Comment by Saran440** [comment]
Fixed it up. Thank you for your advices.

File: account_asset_management_mass_removal/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2020 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Assets Management Mass Removal",
+    "version": "13.0.1.0.0",
+    "license": "AGPL-3",
+    "summary": "Remove Mass Asset",
```
Comment (kittiu) [inline account_asset_management_mass_removal/__manifest__.py:8]: Should be `Remove assets in mass`, if you want summary.

**Pull request closed without merging**